### PR TITLE
feat: support table plugin

### PR DESCRIPTION
**Pull request by zoli**
### Feature Preview

https://github.com/AppFlowy-IO/AppFlowy/assets/3286303/214f6a28-f552-4dc4-815d-6e8748fe1175

---

#### PR Checklist

- [x] My code adheres to the [AppFlowy Style Guide](https://appflowy.gitbook.io/docs/essential-documentation/contribute-to-appflowy/software-contributions/submitting-code/style-guides)
- [ ] I've listed at least one issue that this PR fixes in the description above.
- [x] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [x] All existing tests are passing.

### CODE REVIEW

**Comment by LucasXu0** [comment]
@zoli  I updated the editor dependency to the latest commit and found some new issues.

- The action button will be partially blocked if the table size is larger than the editor size.
Screenshot 2023-08-31 at 11 01 02

- Some cells couldn't be selected. I suppose it's related to the path; the order of the cells may not correspond to their positions.
https://github.com/AppFlowy-IO/AppFlowy/assets/11863087/cf41683c-a6df-4082-bf34-2d154f933d0d

**Comment by LucasXu0** [comment]
Screenshot 2023-09-01 at 13 25 30

@zoli I have disabled some toolbar items in the table, such as "heading" and "bulleted list."

File: frontend/appflowy_flutter/lib/plugins/document/application/editor_transaction_adapter.dart [hunk 1]
```diff
@@ -89,12 +93,17 @@ extension on InsertOperation {
...
-            .expand((element) => element);
-        actions.addAll(childrenActions);
+        for (var i = 0; i < node.children.length; i++) {
+          final n = node.childAtIndexOrNull(i)!;
+          final prevNode = i == 0 ? null : node.childAtIndexOrNull(i - 1);
+          actions.addAll(
+            InsertOperation(currentPath + n.path, [n])
+                .toBlockAction(editorState, previousNode: prevNode),
+          );
+        }
```
[the remark this answers is not in the document]
Comment (zoli) [reply frontend/appflowy_flutter/lib/plugins/document/application/editor_transaction_adapter.dart]: Looks good to me.

**Comment by annieappflowy** [comment]
It would be nice to keep headings and bulleted lists
I understand that they may cause some UI problems at the moment.

**Pull request merged**